x-pack/plugins/fleet/server/services/agents/status.ts

```diff
@@ -92,3 +94,62 @@ export async function getAgentStatusForAgentPolicy(
     events: 0,
   };
 }
+export async function getIncomingDataByAgentsId(
+  esClient: ElasticsearchClient,
+  agentsIds: string[]
+) {
+  try {
+    const searchResult = await esClient.search({
```

Comment:
we had previously some performance issue on a similar query with aggregation in the datastream tab does the `allow_partial_search_results` and `timeout` here prevent this to throw an error?

Reply:
I changed `must` to `filter` and added `size:0` option based on this conversation: https://elastic.slack.com/archives/C0D8ST60Y/p1646929886103879